### DISCUSSION TITLE
Can both openssl1.1.1t and openssl3.0.14 drivers co-exist in a single application/process ?

### DISCUSSION THREAD

**Discussion by GayatriPrasadKovvali** [openssl/openssl#25130] [Q&A]
We have scenarios in our use-case where one of the thread/module in a process uses openssl1 driver that links dynamically (i.e. so/dll) during compile time and if the other thread /module in the same process that loads openssl3 driver(i.e. so/dll)  dynamically using dlopen with RTLD_LOCAL. As per our understanding we may see the issues of symbols corruption between the drivers as openssl3 driver can look up the symbols in the global symbol table(i.e. symbols of openssl1 driver) and can result in abnormal behavior.

**Comment by paulidale** [comment]
That would be my expectation.

I believe some folks do have multiple versions of OpenSSL co-existing but I've no idea how or what issues they run into.
